docs/guides/protect-page-login/flutter-web-redirect.mdx

```diff
@@ -0,0 +1,186 @@
+flutter create myapp
+cd myapp
+```
+
+For HTTP requests we will use the [dio](https://github.com/flutterchina/dio)
+package.
+
+All the packages we use throughout this guide are listed in our `pubspec.yaml`.
+
+```mdx-code-block title="pubspec.yaml"
+import pubspec from '!!raw-loader!../../../code-examples/protect-page-login/flutter_web_redirect/pubspec.yaml'
+import CodeBlock from '@theme/CodeBlock'
+
+<CodeBlock language="yaml">{pubspec}</CodeBlock>
+```
+
+Package sources:
+
+- https://github.com/flutterchina/dio
+- https://github.com/java-james/flutter_dotenv
+
+Next let us create an Authentication service in our `lib/services` directory,
+which we will use to query the Ory APIs for session information.
+
+```shell-session
+touch lib/services/auth.dart
+```
+
+```mdx-code-block title="lib/services/auth.dart"
+import auth from '!!raw-loader!../../../code-examples/protect-page-login/flutter_web_redirect/lib/services/auth.dart'
+
+<CodeBlock language="dart">{auth}</CodeBlock>
+```
+
+Then we need to create an `.env` file in the root of our project to hold the
+`ORY_BASE_URL` variable. This would be the Ory proxy URL (e.g.
+http://localhost:3005/.ory) and the path to the Ory API `/api/kratos/public`.
+
+```shell-session
+touch .env
+```
+
+```mdx-code-block title=".env"
+import env from '!!raw-loader!../../../code-examples/protect-page-login/flutter_web_redirect/env'
+
+<CodeBlock language="text">{env}</CodeBlock>
+```
+
+Finally, we update the `main.dart` file to check for a session cookie on the
+initial load of the application. This will determine if we redirect to the
+sign-in/signup page or not.
+
+```shell-session
+touch lib/main.dart
+```
+
+```mdx-code-block title="lib/main.dart"
+import main from '!!raw-loader!../../../code-examples/protect-page-login/flutter_web_redirect/lib/main.dart'
+
+<CodeBlock language="dart">{main}</CodeBlock>
+```
+
+### Why do I Need the Ory CLI?
+
+```mdx-code-block
+import OryCLI from './_common/ory-cli.mdx'
+
+<OryCLI />
+```
+
+## Run your Flutter Web App
+
+```shell-session
+flutter run -d web-server --web-port 4005
+```
+
+```mdx-code-block
+import SdkEnvVar from '@site/src/components/SdkEnvVar'
+
+<SdkEnvVar />
+```
+
+Start the Ory Cli proxy on [http://localhost:3005](http://localhost:3005) and
+set the proxied application to [http://localhost:4005](http://localhost:4005).
+
+```shell-session
+ory proxy --port 3005 http://localhost:4005
+```
+
+Open the browser on [http://localhost:4000](http://localhost:4000) you should
+now see the Ory managed Login page. Once you have completed the login/signup
+flow you should be redirected back to your Flutter Web App with a session
+cookie. The session cookie can now be sent to your server-side services to
+validate the session on each API call.
+
+:::note
+
+This example only does a simple check on the initial load of the application and
+is not meant to cover complex session checking in Flutter, nor how it would
+interact with your Backend server.
+
+:::
+
```

Comment:
After running application locally it also shows some click counter - is it desired?

Reply:
It's the standard base flutter app. I can remove it if you think it's distracting